fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/MaterializedViewRewriter.java

```diff
@@ -291,12 +300,60 @@ boolean computeCompatibility(OptExpression queryExpr, OptExpression mvExpr) {
         }
     }
 
+    private boolean checkJoinMatch(JoinOperator joinOperator, OptExpression queryExpr, OptExpression mvExpr) {
+        if (joinOperator.isInnerJoin() || joinOperator.isCrossJoin() || joinOperator.isSemiJoin()) {
+            return true;
+        } else if (joinOperator.isLeftOuterJoin() || joinOperator.isLeftAntiJoin()) {
+            // for left outer/anti join, we should make sure the predicates of right child should be equivalent
+            return checkJoinChildPredicate(queryExpr, mvExpr, 1);
```

Comment:
Besides on-predicates, we should make sure other predicates are still equivalent?

This is a bit strict?

eg:
MV: a left join b on a.A = b.A where a.x > 10
Query :  a left join b on a.A = b.A where a.x > 20

This can be rewritten or not?

Reply:
for left outer join, the predicate of right child should be equal. a.x is left table predicate, which can be rewritten